stock_request_purchase/models/purchase_order_line.py

```diff
@@ -12,6 +12,21 @@ class PurchaseOrderLine(models.Model):
         comodel_name="stock.request", string="Stock Requests", copy=False
     )
 
+    def unlink(self):
+        """
+        Cancel the stock.request
+        related to the purchase order line
+        because it does not occur automatically
+        and causes inconsistency by keeping the SR state as 'In Progress' (open).
+        """
+        stock_request_to_cancel = self.env["stock.request"]
```

Comment:
In fact, semantically, this is more correct:

```suggestion
        stock_request_to_cancel = self.env["stock.request"].browse()
```

Reply:
That's not really needed, as it populates the same when you access the model, and if any operation is performed (like addition), a new recordset is populated.

Reply:
Yes, technically it behaves the same. I just warn about semantic